borg/archive.py

```diff
@@ -731,18 +737,24 @@ def rebuild_manifest(self):
 
         Iterates through all objects in the repository looking for archive metadata blocks.
         """
+        def valid_archive(obj):
+            if not isinstance(obj, dict):
+                return False
+            keys = set(obj)
+            return REQUIRED_ARCHIVE_KEYS.issubset(keys)
```

Comment:
I left away the "and keys.issubset(ARCHIVE_KEYS)" because this is used in the case of a lost manifest, so we can't rely on getting the valid archive item keys from the manifest.

If it would be added, we could only use ARCHIVE_KEYS from the source and any additional key in the archive dict would make it invalid and it would get skipped.